### PR TITLE
sync from google3 03-07-2022

### PR DESCRIPTION
**Pull request by devoncarew**
This is a manual sync from the latest from google3. This PR does two things:
- sync to the latest from from google3 as of 03/07/2022
- overwrite the last two commits with correct content (8dd969111167fba7996e0c7d85a5cd03610b59bd and e71438c7bca9be3c1127aeda26ff3b040f5b39fe were introduced unintentionally as part of an incomplete copybara sync)

The new package version - `0.17.10+1` - likely isn't correct from the POV of the number of changes to this package, but I don't think we should re-publish this package in any case before we get a better handle on its state.

### CODE REVIEW

**Comment by devoncarew** [comment]
cc @emmanuel-p

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 ## 0.17.11-dev
+  * Add support for Flutter locale split.
+
+## 0.17.10+2
+  * Allow null safe code when parsing.
+
+## 0.17.10+1
```
Comment (kevmoo) [inline CHANGELOG.md:7]: This was the last version that was published -> https://pub.dev/packages/intl_translation/versions

Might want to clean-up the changelog to remove the 0.17.10+2 release and roll that bit into 0.17.11

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
   * Generate code that passes analysis with `implicit-casts: false`.
+  * Allow use of `MessageExtraction` and `MessageGeneration` without `File`.
```
Comment (kevmoo) [inline CHANGELOG.md:9]: Might want to check the publish package to make sure these changes were actually in that release.

Comment (kevmoo) [reply CHANGELOG.md:9]: Seems like the `arb` bits DID NOT get in that release, seeing as the code is here. But I'm not sure given the borked history

**Review by kevmoo — APPROVED** [review]
The changelog cleanup can happen in the next sync....

**Comment by devoncarew** [comment]
The CI error:

```
Analyzing intl_translation...

  error - lib/extract_messages.dart:34:32 - The method 'latestLanguageVersion' isn't defined for the type 'FeatureSet'. Try correcting the name to the name of an existing method, or defining a method named 'latestLanguageVersion'. - undefined_method

1 issue found.
```

is due to referencing an package:analyzer `FeatureSet.latestLanguageVersion` field that no longer exists. I'm not sure why we're not finding it with our package constraint (`analyzer: '>=0.39.12 <0.40.0'`). The latest published analyzer is `3.3.1`, so we have some work to do here.

**Pull request merged**